### PR TITLE
399-Add regexp validation with error messages for form inputs

### PR DESCRIPTION
**Pull request by klien1**

### CODE REVIEW

**Comment by klien1** [comment]
It would go in form.html and placed where you want the error message to occur. Here's an example.
`<bossy-form-input-validator 
[formGroup]="bossyForm" 
[validators]="element.validate" 
[elementName]="element.name">
</bossy-form-input-validator>`

Element.validator would be an object in the element config. An example element config.
`	element: {
		type: 'email',
		name: 'loginEmail',
		validate: {
			minimum: 3,
			maxmimum: 10,
			presence: true,
			email: true
		}
	}`

File: src/config/form-input-validator.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export class BossyFormInputValidator {
+	private controlName: string;
+	private constraint: BossyFormInputValidatorConfig;
+
+	constructor(controlName: string, validate: BossyFormInputValidatorConfig) {
+		this.controlName = controlName;
+		this.constraint = { [controlName]: validate };
+	}
+	validateElement = (control: FormControl) => {
+		let validate = require('node_modules/validate.js/validate.js');
```
Comment (glawson) [inline src/config/form-input-validator.ts]: This has to be called as an import, like:
import validate from '../../node_modules/validate.js/validate';

File: src/config/form-input-validator.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { FormControl } from '@angular/forms';
+import {BossyFormInputValidatorConfig} from './form-validator';
+
+export class BossyFormInputValidator {
+	private controlName: string;
+	private constraint: BossyFormInputValidatorConfig;
+
+	constructor(controlName: string, validate: BossyFormInputValidatorConfig) {
+		this.controlName = controlName;
+		this.constraint = { [controlName]: validate };
```
Comment (glawson) [inline src/config/form-input-validator.ts]: This line somehow works in code - but should fail.  I would init it with as:
private constraint: BossyFormInputValidatorConfig = new BossyFormInputValidatorConfig();  on line 7, then on line 10:
this.constraint[controlName] = formInputValidatorConfig;  (validate from the constructor needs to be renamed)

File: src/config/form-input-validator.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { FormControl } from '@angular/forms';
+import {BossyFormInputValidatorConfig} from './form-validator';
+
+export class BossyFormInputValidator {
+	private controlName: string;
+	private constraint: BossyFormInputValidatorConfig;
+
+	constructor(controlName: string, validate: BossyFormInputValidatorConfig) {
```
Comment (glawson) [inline src/config/form-input-validator.ts]: rename validate to formInputValidatorConfig

File: src/config/form-input-validator.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+export class BossyFormInputValidator {
+	private controlName: string;
+	private constraint: BossyFormInputValidatorConfig = new BossyFormInputValidatorConfig();
+
+	constructor(controlName: string, formInputValidatorConfig: BossyFormInputValidatorConfig) {
+		this.controlName = controlName;
+		this.constraint[controlName] = formInputValidatorConfig;
+	}
+	validateElement = (control: FormControl) => {
+		const test = validate({[this.controlName]: control.value}, this.constraint);
```
Comment (klien1) [inline src/config/form-input-validator.ts]: I'm not sure why I get this error here.
TS2349:Cannot invoke an expression whose type lacks a call signature.

**Pull request merged**